### ISSUE TITLE
Current focus can be lost if next or previous month buttons become disabled

### ISSUE THREAD

**Issue by mrdaniellewis** [gpbl/react-day-picker#2630]
## To reproduce

```tsx
import { DayPicker } from "react-day-picker";
import "react-day-picker/style.css";

export default function App() {
  return (
    mode="single"
    disabled={{ after: new Date() }}
    endMonth={new Date()}
  );
}
```

CodeSandbox: https://codesandbox.io/p/sandbox/daypicker-bug-report-forked-cff2m5?workspaceId=ws_RGtEKXawCzJL1meZNuGGuQ

Using the keyboard:

1. tab to the previous month button
2. click
3. tab to the next month button
4. click

## Actual Behaviour

The next month button becomes disabled and focus is lost.

## Expected Behaviour

Something should be focused.  I'm not 100% sure the correct pattern:

Option 1: Do not disable the button, mark with `aria-disabled="true"`, and `tabIndex={-1}` to take it out of tab order.  The button will remain focused, but will act as disabled.
Option 2: Focus something nearby, eg the other month navigation button.

https://adrianroselli.com/2023/08/where-to-put-focus-when-deleting-a-thing.html

I'm not sure how much of an issue this is.  I believe it is an issue and it can cause screen-readers to move the current focus to the body throwing the user to the top of the page.  However, I'm not completely sure this is that catastrophic an issue with the latest browsers and screen-readers.

**Comment by rodgobbi** [comment]
@gpbl 
Have you started working on it already?
If not,  I can work on it.
If you already know how to fix this accessibility issue, let me know, otherwise, I'll ask a friend who specializes in accessibility for some directions.
And if you think there are other potential accessibility improvements let me know because I can research more while fixing the issue.

**Comment by gpbl** [comment]
This is indeed an accessibility bug and we have to find a fix for it 🕵 

Can we use `aria-disabled` instead of `disabled`? These two attributes have been causing some trouble in DayPicker, and I don't understand the real difference between them. 

On the other side, this could be a breaking change. Developers using the `[disabled]` selector in their CSS would need to change it to `[aria-disabled="true"]`.

**Comment by rodgobbi** [comment]
@gpbl 
Besides the breaking change, I think it also makes sense for the button to be `disabled`, for all types of users, and we get the correct accessibility by default from the browser.

I'll research how to handle the focus for this type of use case, of disabling the element focused on the page, and share what I learn here.

**Comment by gpbl** [comment]
Some ideas :

- quick workaround (cumbersome and to verify); by passing a callback to `onMonthChange` prop, dev should be able to focus an element in case the month is the last/first
- have a `focusFallbackEl` prop to use in this specific case (should have a list of testable cases): focus the "first focusable thing" inside the root container, if not existing, use the focus fallback

**Comment by gpbl** [comment]
I believe we can use `aria-disabled` instead: see https://github.com/gpbl/react-day-picker/pull/2685.
